app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -292,7 +294,14 @@ class PatientSummaryScreenController @Inject constructor(
         .ofType<PatientSummaryRestoredWithBPSaved>()
         .map { it.wasBloodPressureSaved }
 
-    val mergedBpSaves = Observable.merge(bloodPressureSaves, bloodPressureSaveRestores)
+    val allBpsForPatientDeleted = patientUuids
+        .switchMap(bpRepository::bloodPressureCount)
+        .map { recordedBpCount -> recordedBpCount == 0 }
+
+    val shouldShowAppointmentSheet = Observable.merge(bloodPressureSaves, bloodPressureSaveRestores)
```

Comment:
There is a bug in this logic here.
If the stream `Observable.merge(bloodPressureSaves, bloodPressureSaveRestores)` does not emit at all then this entire chain will become inactive because you are using `withLatestFrom`. This will lead to save and back clicks not being handled at all.

To reproduce this case:
- Open a patient. 
- Open and close the BP sheet without making any changes.
Now if you try to press `Save` or `Back`, it won't work.
 
I tried to fix this but this chain will have to be rewritten.

Reply:
🥇

Reply:
Alright, I took a look at it and trying to fix the current stream seems a little complex. But we anyway have to make it so that the appointment sheet has to be shown always, regardless of whether the BP was saved or not.

I'll make this change in this PR and remove this problem in one stroke.

Reply:
Never mind, I found an easier fix. Pushing the fix now.